### PR TITLE
Start using Python 3.10

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
This PR adds using Python 3.10 in CI. 

- [X] Have you provided a meaningful PR description?

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
The PR changes workflow for building with Intel/LLVM open-source compiler to using Python 3.10, documentation generation and coverage workflows to building with Python 3.10. 

conda-package workflow builds dpctl for Python 3.10 for Linux, but not for Windows (unsatisfied dependencies).

File: .github/workflows/conda-package.yml [hunk 1]
```diff
@@ -324,7 +321,7 @@ jobs:
     runs-on:  ${{ matrix.runner }}
     strategy:
       matrix:
-        python: [3.9]
+        python: ['3.9', '3.10']
```
Comment (AndresGuzman-Ballen) [inline .github/workflows/conda-package.yml:324]: Did you intentionally leave out '3.8' here?

**Review by AndresGuzman-Ballen — APPROVED** [review]
Other than that '3.8' comment, LGTM!

File: .github/workflows/conda-package.yml (hunk 1 above)
Comment (AndresGuzman-Ballen) [reply .github/workflows/conda-package.yml]: Sounds good!

**Pull request merged**